Python/_warnings.c

```diff
@@ -520,14 +525,20 @@ warn_explicit(PyObject *category, PyObject *message,
         goto cleanup;
     }
 
+    if (_PyUnicode_EqualToASCIIString(action, "ignore")) {
+        goto return_none;
+    }
+
     /* Store in the registry that we've been here, *except* when the action
        is "always". */
     rc = 0;
     if (!_PyUnicode_EqualToASCIIString(action, "always")) {
         if (registry != NULL && registry != Py_None &&
-                PyDict_SetItem(registry, key, Py_True) < 0)
+                PyDict_SetItem(registry, key, Py_True) < 0) {
             goto cleanup;
-        else if (_PyUnicode_EqualToASCIIString(action, "ignore"))
+        }
+
+        if (_PyUnicode_EqualToASCIIString(action, "ignore"))
```

Comment:
Isn't this already tested above?

Reply:
I removed it. I kept it for tests, but then I forgot to remove it.